File: src/main/java/com/example/dpop/id_fsc/IdFscService.java
package com.example.dpop.id_fsc;

import com.example.dpop.id_fsc.internal.FscCodeRepository;
import org.springframework.stereotype.Service;

import java.time.Instant;
import java.util.Optional;

@Service
public class IdFscService {

    private final FscCodeRepository fscCodeRepository;

    public IdFscService(FscCodeRepository fscCodeRepository) {
        this.fscCodeRepository = fscCodeRepository;
    }

    public String identify() {
        return "id_fsc: identity verified";
    }

    public boolean hasValidFsc(Long personId) {
        return fscCodeRepository.findByPersonId(personId)
                .map(fscCode -> fscCode.isValid())
                .orElse(false);
    }

    public boolean validateFsc(Long personId, String code) {
        return fscCodeRepository.findByPersonIdAndCode(personId, code)
                .map(fscCode -> fscCode.isValid())
                .orElse(false);
    }

    public void createFsc(Long personId, String code, Instant expiresAt) {
        fscCodeRepository.save(new com.example.dpop.id_fsc.internal.FscCode(personId, code, expiresAt));
    }
}

File: src/main/kotlin/com/example/dpop/id_fsc/IdFscService.kt
package com.example.dpop.id_fsc

import com.example.dpop.id_fsc.internal.FscCode
import com.example.dpop.id_fsc.internal.FscCodeRepository
import org.springframework.stereotype.Service
import java.time.Instant

@Service
class IdFscService(private val fscCodeRepository: FscCodeRepository) {

    fun identify(): String = "id_fsc: identity verified"

    fun hasValidFsc(personId: Long): Boolean =
        fscCodeRepository.findByPersonId(personId)
            .map { it.isValid }
            .orElse(false)

    fun validateFsc(personId: Long, code: String): Boolean =
        fscCodeRepository.findByPersonIdAndCode(personId, code)
            .map { it.isValid }
            .orElse(false)

    fun createFsc(personId: Long, code: String, expiresAt: Instant) {
        fscCodeRepository.save(FscCode(personId, code, expiresAt))
    }
}
